x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -171,3 +195,38 @@ impl<K> BorshSerialize for CallProgramArgs<'_, K> {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{DeferDeserialize, ExternalCallError};
+
+    #[test]
+    fn defer_ok_variant() {
+        let bytes = DeferDeserialize(vec![1, 0]);
+        assert_eq!(bytes.deserialize::<Result<_, ()>>().unwrap(), Ok(false));
+    }
+
+    #[test]
+    fn defer_err_variant() {
+        let bytes = DeferDeserialize(vec![0, 1]);
+        assert_eq!(
+            bytes
+                .deserialize::<Result<(), ExternalCallError>>()
+                .unwrap(),
+            Err(ExternalCallError::CallPanicked)
+        );
+    }
+
+    #[test]
+    fn defer_no_len_prefix() {
+        let inner_bytes = vec![1, 4, 3, 2, 1];
+        let defer_res = borsh::from_slice::<Result<DeferDeserialize, ()>>(&inner_bytes);
+        assert!(defer_res.is_ok());
+        let Ok(defer) = defer_res else {
+            unreachable!();
+        };
+        assert!(defer.is_ok_and(|bytes| bytes.0 == vec![4, 3, 2, 1]));
+        let defer = borsh::from_slice::<Result<Vec<u8>, ExternalCallError>>(&inner_bytes);
+        assert!(defer.is_err());
+    }
```

Comment:
The test here is overcomplicated. Why don't you do something more like this:
```rust
type ExptectedType = u64;

let expected: ExpectedType = 42;
let serialized = borsh::to_vec(expected).unwrap();
let deferred: DeferDeserialize = borsh::from_slice(&serialized);
assert_eq!(deferred.0 == serialized);
let actual = deferred.deserialize::<ExpectedType>().unwrap();
assert!(actual, expected);
```
(I didn't actual compile and run this, but it should be something this simple).

Reply:
True, well I just removed other tests and replaced them by yours because they were not properly testing the `DeferDeserialize` struct but felt duplicated.